src/Core/Services/Implementations/EventService.cs

```diff
@@ -349,8 +349,29 @@ public async Task LogProviderUsersEventAsync(IEnumerable<(ProviderUser, EventTyp
         await _eventWriteService.CreateAsync(e);
     }
 
+    public async Task LogProviderOrganizationEventsAsync(Guid providerId, IEnumerable<ProviderOrganization> providerOrganizations,
+        EventType type, DateTime? date = null)
+    {
+        var providerAbilities = await _applicationCacheService.GetProviderAbilitiesAsync();
+        if (!CanUseProviderEvents(providerAbilities, providerId))
+        {
+            return;
+        }
+
+        var messages = providerOrganizations.Select(po => new EventMessage(_currentContext)
+        {
+            ProviderId = po.ProviderId,
+            ProviderOrganizationId = po.Id,
+            Type = type,
+            ActingUserId = _currentContext?.UserId,
+            Date = date.GetValueOrDefault(DateTime.UtcNow)
+        });
```

Comment:
This assumes that the `providerId` argument is the same as the `ProviderId` of each `ProviderOrganization` in the `IEnumerable`, but that's not guaranteed. The `CanUseProviderEvents` check needs to be in the loop to check for each `ProviderOrganization.ProviderId` separately, and you can remove the `providerId` argument.

More generally, this interface should fit the pattern for bulk events elsewhere in this file: it should take an `IEnumerable<(ProviderOrganization, EventType, DateTime?)>` argument. See the bulk organization user events method here for an example: https://github.com/bitwarden/server/blob/master/src/Core/Services/Implementations/EventService.cs#L248

Then you can make the `LogProviderOrganizationEventAsync` method (for a single event) call this method so we're not duplicating the logic.

Reply:
That makes perfect sense, I've implemented it like you suggested.